controllers/hardwareclassification_controller.go

```diff
@@ -17,36 +14,222 @@ package controllers
 
 import (
 	"context"
+	"errors"
+	"strconv"
 
 	"github.com/go-logr/logr"
+
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+
+	hwcc "hardware-classification-controller/api/v1alpha1"
+
+	bmh "github.com/metal3-io/baremetal-operator/pkg/apis/metal3/v1alpha1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
-
-	metal3iov1alpha1 "hardware-classification-controller/api/v1alpha1"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/source"
 )
 
+var validHostList []bmh.BareMetalHost
```

Comment:
We're almost never going to want global variables for controllers. Each controller can run multiple goroutines, reconciling multiple resources in parallel. We don't want those goroutines to step on each other. If this is working for you locally that is because the default is to only run 1 goroutine, but that is not how I would expect the controller to run in a production setting.

I would move all of these variables into the functions that use them.

Reply:
Yes  we have removed use of global variables. 
Addressed in new commit: c239dc6a2734041cdaa3cd678704237331a80b83